api/src/permissions/modules/fetch-global-access/lib/fetch-global-access-for-roles.ts

```diff
@@ -1,9 +1,17 @@
+import type { Accountability } from '@directus/types';
 import { getSimpleHash } from '@directus/utils';
-import type { Knex } from 'knex';
+import type { Context } from '../../../types.js';
 import type { GlobalAccess } from '../types.js';
 import { fetchGlobalAccessForQuery } from '../utils/fetch-global-access-for-query.js';
 
-export async function fetchGlobalAccessForRoles(knex: Knex, roles: string[]): Promise<GlobalAccess> {
-	const query = knex.where('role', 'in', roles);
-	return await fetchGlobalAccessForQuery(query, `global-access-roles-${getSimpleHash(JSON.stringify(roles))}`);
+export async function fetchGlobalAccessForRoles(
+	accountability: Pick<Accountability, 'roles' | 'ip'>,
+	context: Pick<Context, 'knex'>,
+): Promise<GlobalAccess> {
+	const query = context.knex.where('role', 'in', accountability.roles);
+	return await fetchGlobalAccessForQuery(
+		getSimpleHash(`${accountability.roles}-${accountability.ip}`), // Hash it so we dont expose user IPs
+		query,
+		accountability.ip,
```

Comment:
Any reason to still rely on the custom caching logic in `fetchGlobalAccessForQuery`? I think we could simply do something very similar to the rest of the `fetch*` functions where we just use the `withCache` util, since the accountability is out cache key here, no?

This would simplify `fetchGlobalAccessForQuery` and these two functions.